### PR TITLE
wifi: Add WPA-PSK support for Wi-Fi management extensions

### PR DESCRIPTION
**Pull request by kapbh**

### CODE REVIEW

**Review by sachinthegreen — APPROVED** [review]

**Review by VivekUppunda — APPROVED** [review]

**Review by sachinthegreen — COMMENTED** [review]
This might need a `Changelog` entry.

**Review by krish2718 — APPROVED** [review]

**Review by richabp — COMMENTED** [review]
Add a Changelog entry, if required.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -538,6 +538,7 @@ Wi-Fi samples
 
   * Modified ``connect`` command to provide better control over connection parameters.
   * Added ``Auto-Security-Personal`` mode to the ``connect`` command.
+  * Added support for the ``WPA-PSK`` security mode to the wifi_mgmt_ext.
```
Comment (richabp) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:541]:
```suggestion
  * Added support for the ``WPA-PSK`` security mode to the ``wifi_mgmt_ext`` library.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 1 above)
Comment (kapbh) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Done

**Review by rado17 — APPROVED** [review]

**Pull request merged**